example/plasticity/plasticity.cpp

```diff
@@ -57,29 +62,39 @@ struct recipe: public arb::recipe {
         if (gid == 0) return arb::spike_source_cell{src, arb::regular_schedule(f_spike)};
         // all others are receiving cable cells; single CV w/ HH
         arb::segment_tree tree; tree.append(arb::mnpos, {-r_soma, 0, 0, r_soma}, {r_soma, 0, 0, r_soma}, 1);
-        auto decor = arb::decor{};
-        decor.paint(all, arb::density("hh", {{"gl", 5}}));
-        decor.place(center, arb::synapse("expsyn"), syn);
-        decor.place(center, arb::threshold_detector{-10.0}, det);
-        decor.set_default(arb::cv_policy_every_segment());
+        auto decor = arb::decor{}
+            .paint(all, arb::density("hh", {{"gl", 5}}))
+            .place(center, arb::synapse("expsyn"), syn)
+            .place(center, arb::threshold_detector{-10.0}, det)
+            .set_default(arb::cv_policy_every_segment());
         return arb::cable_cell({tree}, {}, decor);
     }
 };
 
+// For demonstration: Avoid interleaving std::cout in multi-threaded scenarios.
+// NEVER do this in HPC!!!
+std::mutex mtx;
+
 void sampler(arb::probe_metadata pm, std::size_t n, const arb::sample_record* samples) {
     auto* loc = arb::util::any_cast<const arb::mlocation*>(pm.meta);
-    std::cout << std::fixed << std::setprecision(4);
+
     for (std::size_t i = 0; i<n; ++i) {
+        std::lock_guard<std::mutex> lock{mtx};
         auto* value = arb::util::any_cast<const double*>(samples[i].data);
-        std::cout << "|  " << samples[i].time << " |      " << loc->pos << " | " << *value << " |\n";
+        std::cout << std::fixed << std::setprecision(4)
+                  << "|  " << samples[i].time << " |      " << loc->pos << " | " << *value << " |\n";
```

Comment:
would be good practice to reset the `cout` state to what it was before, after this line (I know it is not important here, though).

Reply:
It sadly brings lots of hassle, but I'll add something ;)

Reply:
not strictly needed for this PR